kvui.py

```diff
@@ -299,6 +300,47 @@ def apply_selection(self, rv, index, is_selected):
         self.selected = is_selected
 
 
+class AutocompleteHintInput(TextInput):
+    min_chars = NumericProperty(3)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.dropdown = DropDown()
+        self.dropdown.bind(on_select=lambda instance, x: setattr(self, 'text', x))
+        self.bind(on_text_validate=self.on_message)
+
+    def on_message(self, instance):
+        App.get_running_app().commandprocessor("!hint "+instance.text)
+
+    def on_text(self, instance, value):
+        if len(value) >= self.min_chars:
+            self.dropdown.clear_widgets()
+            ctx: context_type = App.get_running_app().ctx
+            item_names = ctx.item_names._game_store[ctx.game].values()
```

Comment:
The lack of typing in kvui.py is a bit worrying, but kivy not having stubs definitely makes it hard to do it properly.

This fixes the typing issue mentioned above. ctx.game is optional and if it's missing there is nothing to show anyway. If you are sure that ctx.game can never be None when on_text happens, you could also assert instead.
```suggestion
            if not ctx.game:
                return
            item_names = ctx.item_names._game_store[ctx.game].values()
```